include/seqan3/core/type_traits/deferred_crtp_base.hpp

```diff
@@ -47,6 +48,39 @@ struct deferred_crtp_base
     using invoke = crtp_base<derived_t, args_t...>;
 };
 
+/*!\brief An invocable wrapper that defers the instantiation of a crtp_base class.
+ * \ingroup type_traits
+ * \tparam crtp_base The crtp base class to be deferred. Must be a template template parameter.
+ * \tparam args      A non-type template parameter pack used to augment the `crtp_base` class.
+ *
+ * \details
+ *
+ * This transformation trait wrapper allows to defer the template instantiation of crtp-base classes. This can be useful
+ * if the crtp_base class should be augmented with traits or other templates, especially when using variadic
+ * crtp_bases. The help function seqan3::detail::invoke_deferred_crtp_base can be used to instantiate the
+ * deferred crtp base with the respective derived type. This class wrapper can only be used to augment a crtp-base
+ * class with non-type arguments. See seqan3::detail::deferred_crtp_base for a version that accepts type template
+ * parameters.
+ *
+ * ### Example
+ *
+ * The following snippet demonstrates the use of the deferred crtp base class instantiation.
+ *
+ * \include test/snippet/core/type_traits/deferred_crtp_base.cpp
+ *
+ * \see seqan3::detail::invoke_deferred_crtp_base
+ * \see seqan3::detail::deferred_crtp_base
+ */
+template <template <typename, auto ...> typename crtp_base, auto ...args>
+struct deferred_crtp_base_vargs
```

Comment:
I haven't found this applied in this PR, so I think it is technically unrelated to this PR, but it looks fine.

Reply:
It was related at some point, but the code changed and I didn't need it anymore, but figured it might be still useful in other places. So we have it already.